File: Data Structures & Algorithms/longest-turbulent-subarray/submission-1.java
class Solution {
    public int maxTurbulenceSize(int[] arr) {
        int m1=1,m2=1;
        int l1=1,l2=1;
        for(int i=0;i<arr.length-1;i++){
            if(i%2==0){
                if(arr[i]<arr[i+1]){
                    l2=1;
                    l1++;
                }
                else if(arr[i]>arr[i+1]){
                    l2++;
                    l1=1;
                }
                else{
                    l1=1;
                    l2=1;
                }

            }
            else{
                if(arr[i]>arr[i+1]){
                    l2=1;
                    l1++;
                }
                 else if(arr[i]<arr[i+1]){
                    l1=1;
                    l2++;
                }
                else{
                    l1=1;
                    l2=1;
                }

            }
            m1=Math.max(l1,m1);
            m2=Math.max(l2,m2);

        }
        return Math.max(m1,m2);
        
    }
}

File: Data Structures & Algorithms/network-delay-time/submission-1.js
class Solution {
    /**
     * @param {number[][]} times
     * @param {number} n
     * @param {number} k
     * @return {number}
     */
    networkDelayTime(times, n, k) {
    const map = new Map();

for (const [source, target, weight] of times) {
  map.set(source, map.get(source) ?? []);
  map.get(source).push([target, weight]);
}
const queue=[];
const dist= new Array(n + 1).fill(Infinity);
dist[k]=0;
queue.push(k);
while(queue.length>0){
    const node = queue.shift();
    for(const [adjnode,time] of map.get(node)??[]){
        if(dist[adjnode]>dist[node]+time){
        dist[adjnode]=dist[node]+time;
        queue.push(adjnode);
        }
    }
}
const ans = Math.max(...dist.slice(1));
 return ans==Infinity ?-1:ans;
    }
}
